smctemp.cc

```diff
@@ -366,6 +382,22 @@ kern_return_t SmcAccessor::PrintAll() {
   return kIOReturnSuccess;
 }
 
+double SmcTemp::CalculateAverageTemperature(const std::vector<std::string>& sensors,
+                                     const std::pair<unsigned int, unsigned int>& limits) {
+  double temp = 0.0;
+  size_t valid_sensor_count = 0;
+  for (auto sensor : sensors) {
+    auto sensor_value = smc_accessor_.ReadValue(sensor.c_str());
+    if (sensor_value >= limits.first &&
+        sensor_value <= limits.second) {
+      temp += sensor_value;
+      valid_sensor_count++;
+    }
+  }
+  temp /= valid_sensor_count;
```

Comment:
If all sensor values are out of limits, zero-devided will happen.
```suggestion
  if (valid_sensor_count > 0) {
    temp /= valid_sensor_count;
  }
```